flake8_trio.py

```diff
@@ -41,6 +31,79 @@ class Statement(NamedTuple):
     "TRIO110": "`while <condition>: await trio.sleep()` should be replaced by a `trio.Event`.",
 }
 
+
+class Statement(NamedTuple):
+    name: str
+    lineno: int
+    col_offset: int = 0
+
+    # ignore col offset since many tests don't supply that
+    def __eq__(self, other: Any) -> bool:
+        return isinstance(other, Statement) and (self.name, self.lineno) == (
+            other.name,
+            other.lineno,
+        )
+
+
+HasLineInfo = Union[ast.expr, ast.stmt, ast.arg, ast.excepthandler, Statement]
+
+
+class TrioScope:
+    def __init__(self, node: ast.Call, funcname: str):
+        self.node = node
+        self.funcname = funcname
+        self.variable_name: Optional[str] = None
+        self.shielded: bool = False
+        self.has_timeout: bool = False
```

Comment:
I'd default this to `True`, then set it to `False` inside the if-CancelScope block and drop the else.  Also add a comment that handling `scope.shield = True` is done elsewhere.